vork/src/main/scala/vork/markdown/vorkParserExtension.scala

```diff
@@ -23,17 +23,26 @@ class VorkParserExtension(options: Options) extends Parser.ParserExtension {
       * Creating a custom node extending `ContentNode` or `CustomNode` did not help either.
       */
     override def createNode(nodeChars: BasedSequence): ast.Node = {
-      val key = nodeChars.midSequence(2, nodeChars.length() - 1).trim()
-      val value = site.getOrElse(key.unescape(), sys.error(s"Missing '$key' site variable."))
-      new ast.Text(PrefixedSubSequence.of(value, nodeChars).removeSuffix(nodeChars))
+      nodeChars.toString match {
+        case VariableInjectionPattern(key) =>
+          val value = site.getOrElse(key, sys.error(s"Missing '$key' site variable."))
+          new ast.Text(PrefixedSubSequence.of(value, nodeChars).removeSuffix(nodeChars))
+        case _ =>
+          sys.error("Flexmark matched a variable injection which is not of the expected shape.")
+      }
     }
 
+    private final val totalLength = document.getTextLength
+    private final val VariableInjectionPattern = "!\\[([^\\]\\[]*)\\]".r
```

Comment:
> implemented using standard markdown AST processing as opposed to regex search and replace.

😅  update readme?

Reply:
discussed offline, we're still using the AST processing since the regex only applies for an AST node that represents a link and whose shape we already know :smile:

Reply:
Nevermind, we still use AST processing machinery!